docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,121 @@
+- DAML React libraries
+- quick iteration against the :ref:`DAML Ledger Sandbox <sandbox-manual>`.
+- authentication
+- deploy your application in the cloud as a Docker container
 
-Event-driven applications
-=========================
+It provides an excellent starting point to iterate on, so you can build your own application.
 
-**Event-driven** applications react to events on the the ledger and generate commands and other outputs on a per-event basis. They do not require access to ledger state beyond the event they are reacting to.
 
-Examples are sink applications that read the ledger and dump events to an external store (e.g. an external (reporting) database).
+Frontend
+~~~~~~~~
 
-State-driven applications
-=========================
+We recommended building your frontend with the `React <https://reactjs.org>`_ framework. We provide
+several support libraries to build your React frontend for a DAML application.
 
-**State-driven** applications build up a real-time view of the ledger state by reading events and recording contract create and archive events. They then generate commands based on a given state, not just single events.
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| Name                                                         | Summary                                                                  |
++==============================================================+==========================================================================+
+| `@daml/react <https://www.npmjs.com/package/@daml/react>`_   | React hooks to query/create/exercise DAML contracts                      |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/ledger <https://www.npmjs.com/package/@daml/ledger>`_ | DAML ledger object to connect and directly submit commands to the ledger |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
+| `@daml/types  <https://www.npmjs.com/package/@daml/types>`_  | DAML data types                                                          |
++--------------------------------------------------------------+--------------------------------------------------------------------------+
 
-Examples of these are automation and interactive applications that let a user or code react to complex state on the ledger (e.g. the DA Navigator tool).
+You can install any of these libraries by running ``yarn add @daml/react`` in the ``ui`` directory
+of your project. Please explore the ``create-daml-app`` example project to see the usage of these
+libraries.
 
-Which approach to take
-======================
+The interface between the frontend and the backend of your application is defined by your DAML code.
```

Comment:
This makes it sound like DAML is some sort of an interface or middleware language.  Why not move it to the backend side in the docs and explain everything there? If you run the sandbox it runs with your dar file. IMHO makes it easier to understand.

Reply:
Hmm, I think it kind of is an interface language. It tells you how data is represented and how it can be changed and by whom. I don't want to move this to the backend, because otherwise I'd have to talk about the frontend in the backend section.

Reply:
@bame-da @hurryabit Can we have a consensus here on how we talk about DAML with respect to frontend and backend? At the moment in the recommended architecture it's represented as an interface language. If that's the way to go, fine. I just think it's underselling what it is and should reside on the backend side.

Reply:
I have changed the diagram and this sentence so that hopefully everybody is happy.

Reply:
added final changes by @nemanja-da.